### PR TITLE
Support adding caches to bookmark lists (fix #118)

### LINKED ISSUE #118: Support adding caches to bookmark lists
The geocaching.com web interface provides the function "Bookmark listing", which I am using e.g. to add a cache to my FTF list right after logging the FTF.
It should be possible to do this from c:geo (because in the evening at home I might forget to do this).

### PR DESCRIPTION
**Pull request by fm-sys**
Only available for GC premium members - fix #118

grafik

grafik
grafik
grafik

### CODE REVIEW

**Comment by fm-sys** [comment]
TODO: support access from cache details activity

File: main/res/values/strings.xml [hunk 1]
```diff
@@ -410,6 +410,7 @@
     <string name="caches_remove_all_completely">Delete all from device</string>
     <string name="caches_remove_selected_completely">Delete selected from device</string>
     <string name="caches_delete_events">Delete past events</string>
+    <string name="caches_upload_bookmarklist">Upload cache(s) to bookmark list</string>
```
Comment (Lineflyer) [inline main/res/values/strings.xml:413]: As you are inside the list menu, how about only "Upload to bookmark list" or (to make clear its about the website) "Add to website bookmark list"

**Comment by Lineflyer** [comment]
Does it handle if the list contains caches not belonging to GC (UDC, other connectors)?

**Comment by fm-sys** [comment]
> Does it handle if the list contains caches not belonging to GC (UDC, other connectors)?

Yes these caches will be ignored... See https://github.com/cgeo/cgeo/blob/75f49c3dc110f9437c54024bb48213d52981a5cf/main/src/cgeo/geocaching/connector/gc/GCParser.java#L1073-L1076

File: main/res/values/strings.xml (hunk 1 above)
Comment (fm-sys) [reply main/res/values/strings.xml]: Done

File: main/res/values/strings.xml [hunk 2]
```diff
@@ -1564,6 +1565,12 @@
     <string name="search_bookmark_list">Bookmark list</string>
     <string name="search_pocket_is_new">new</string>
     <string name="search_pocket_and_bookmark_loading">Loading a list of Pocket Queries and bookmark lists</string>
+    <string name="search_bookmark_loading">Loading bookmark lists</string>
+    <string name="search_bookmark_select">Select bookmark lists</string>
+    <string name="search_bookmark_new">Create new bookmark lists</string>
```
Comment (Lineflyer) [inline main/res/values/strings.xml:1570]: Should probably be singular "Create new bookmark list"

File: main/res/values/strings.xml [hunk 3]
```diff
@@ -1564,6 +1565,12 @@
...
+    <string name="search_bookmark_adding_caches_success">All GC geocaches where added to the selected bookmark list</string>
```
Comment (Lineflyer) [inline main/res/values/strings.xml:1571]: "were" instead of "where" or even better "have been"

File: main/res/values/strings.xml (hunk 2 above)
Comment (fm-sys) [reply main/res/values/strings.xml]: Thanks, good catch... Fixed.

File: main/res/values/strings.xml (hunk 3 above)
Comment (fm-sys) [reply main/res/values/strings.xml]: You're totally right. That happens when you focus more on code architecture, than strings... Solved as well.

**Comment by fm-sys** [comment]
Let's merge to master to have another cool feature for our upcoming beta. The code is completely decoupled, so at least no regressions are to be expected...

**Pull request merged**